examples/notebooks/get-started/learn_beam_basics_by_doing.ipynb

```diff
@@ -0,0 +1,1070 @@
+        "> Each element goes into its own line in the output files."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "D80HrbeKFxCv"
+      },
+      "source": [
+        "Now, you can try it. Save the results to a file path prefix `'output'` and make the file_name_suffix `'.txt'`"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "v6RvVVINKOt9"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Edit This Code\n",
+        "inputs_pattern = 'data/SMSSpamCollection'\n",
+        "\n",
+        "# Remember: | is the apply function in Beam in Python\n",
+        "outputs = (\n",
+        "    pipeline\n",
+        "      | 'Take in Dataset' >> beam.io.ReadFromText(inputs_pattern)\n",
+        "      | 'Write results' >> beam.io.WriteToText(..., file_name_suffix = ...)\n",
+        "      # To see the results\n",
+        "      | 'Print results' >> beam.Map(print)\n",
+        ")\n",
+        "\n",
+        "# To run the pipeline\n",
+        "pipeline.run()\n",
+        "\n",
+        "# The command used to view your output txt file.\n",
+        "# If you choose to save the file path prefix to a different location or change the file type,\n",
+        "# you have to update this command as well.\n",
+        "! head output*.txt"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "SjphdAsmOQlh"
+      },
+      "source": [
+        "> *`!` means \"run\" for Apache Beam*"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "nRYxclPaJArp"
+      },
+      "outputs": [],
+      "source": [
+        "#@title Answer\n",
+        "inputs_pattern = 'data/SMSSpamCollection'\n",
+        "\n",
+        "outputs = (\n",
+        "    pipeline\n",
+        "      | 'Take in Dataset' >> beam.io.ReadFromText(inputs_pattern)\n",
+        "      # ADDED\n",
+        "      | 'Write results' >> beam.io.WriteToText(\"ansoutput1\", file_name_suffix = \".txt\")\n",
+        "      | 'Print results' >> beam.Map(print)\n",
+        ")\n",
+        "\n",
+        "pipeline.run()\n",
+        "\n",
+        "# The file this data is saved to is called \"ansoutput1\" as seen in the WriteToText transform.\n",
+        "# The command below and the transform input should match.\n",
+        "! head ansoutput1*.txt"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "iie23Id-O33B"
+      },
+      "source": [
+        "# 2. PTransforms\n",
+        "\n",
+        "Now that we have read in our code, we can now process our data and clean it up. Remember that the purpose of this pipeline is to **clean up our data to derive features** to use in a classifier that detects spam emails from non-spam (ham) SMS.\n",
```

Comment:
It would be good to clarify the intent of cleanup (what exactly we plan to do).

Reply:
Polished it a little bit by changing the purpose to data exploration. Do not want to mention features here since it could be a new concept for beginners.